### PR TITLE
[#6198]: Add QR code scanner support in extensions panel

### LINKED ISSUE #6198: Add QR code scanner support in extensions panel
### User Story

As a user, I want an easy way to scan extensions QR code in the `extensions` panel

### Description

*Type*: Feature

### PR DESCRIPTION
**Pull request by alwx**
Fixes #6198

### CODE REVIEW

**Comment by ghost** [comment]
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status.html))?

**Review by jeluard — APPROVED** [review]

File: src/status_im/extensions/core.cljs [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
 
 (defn url->storage-details [s]
   (when s
     (let [[_ type id] (string/split s #".*[:/]([a-z]*)@(.*)")]
       [(keyword type) id])))
+
+(fx/defn set-extension-url-from-qr
+  [{:keys [db]} url]
+  {:db       (assoc db :extension-url url)
+   :dispatch [:navigate-back]})
```
Comment (janherich) [inline src/status_im/extensions/core.cljs]: Unnecessary dispatch, should be just `navigate-back` function from navigation ns.

**Review by janherich — DISMISSED** [review]

File: src/status_im/extensions/core.cljs (hunk 1 above)
Comment (alwx) [reply src/status_im/extensions/core.cljs]: I don't agree.

`:navigate-back` is the only handler we use in this case, and it also has a check for a modal dialog. The function you're talking about is a side-effect, it's a strange idea to just call it like that. 

Maybe a good way of solving this is to add an fx and do something like 
```
{:db db
 :navigation/navigate-back nil}
```
But I don't think it's something related to this issue.

**Review by flexsurfer — APPROVED** [review]

**Comment by jeluard** [comment]
@alwx Please resolve the conflict ! :)

File: src/status_im/extensions/core.cljs (hunk 1 above)
Comment (yenda) [reply src/status_im/extensions/core.cljs]: it does the same as :navigate-back does without an unnecessary dispatch there is no side-effect it updates the db and adds the ::navigate-back nil effect

Comment (yenda) [reply src/status_im/extensions/core.cljs]:
```
(fx/defn navigate-back
  [{{:keys [navigation-stack view-id] :as db} :db}]
  (assoc
   {::navigate-back nil}
   :db (let [[previous-view-id :as navigation-stack'] (pop navigation-stack)
             first-in-stack (first navigation-stack)]
         (if (= view-id first-in-stack)
           (-> db
               (assoc :view-id previous-view-id)
               (assoc :navigation-stack navigation-stack'))
           (assoc db :view-id first-in-stack)))))

(handlers/register-handler-fx
 :navigate-back
 (re-frame/enrich -preload-data!)
 (fn [cofx _]
   (navigate-back cofx)))
```

Comment (alwx) [reply src/status_im/extensions/core.cljs]: Then sorry, I simply looked at the wrong function

**Review by yenda — APPROVED** [review]

**Comment by asemiankevich** [comment]
@alwx @jeluard 

what should be shown for user when i scan extension URL qr code within qr scanner from Home screen? (Home - plus button - scan QR)? 
If i do this now in this PR - nothing is shown for user, however would be nice to say something (we have some error dialog for this). WDYT? 

image

**Comment by asemiankevich** [comment]
so this PR implements scanning universal links for extensions only. No validation is added (future PRs per @jeluard )

**Pull request merged**